packages/openapi-specification/openapi.yaml

```diff
@@ -1,742 +1,177 @@
   license:
     name: Apache-2.0
     url: 'https://www.apache.org/licenses/LICENSE-2.0.html'
+externalDocs:
+  description: More information about Webcrawler API
+  url: https://webik.ms.mff.cuni.cz/nswi153/seminar-project-webcrawler.html
 tags:
-  - name: pet
-    description: Everything about your Pets
-  - name: store
-    description: Access to Petstore orders
-  - name: user
-    description: Operations about user
+  - name: website-record
+    description: Record of scheduled website crawls.
+
 paths:
-  /pet:
-    post:
-      tags:
-        - pet
-      summary: Add a new pet to the store
-      description: ''
-      operationId: addPet
-      responses:
-        '200':
-          description: successful operation
-          content:
-            application/xml:
-              schema:
-                $ref: '#/components/schemas/Pet'
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Pet'
-        '405':
-          description: Invalid input
-      security:
-        - petstore_auth:
-            - 'write:pets'
-            - 'read:pets'
-      requestBody:
-        $ref: '#/components/requestBodies/Pet'
-    put:
-      tags:
-        - pet
-      summary: Update an existing pet
-      description: ''
-      operationId: updatePet
-      externalDocs:
-        url: "http://petstore.swagger.io/v2/doc/updatePet"
-        description: "API documentation for the updatePet operation"
-      responses:
-        '200':
-          description: successful operation
-          content:
-            application/xml:
-              schema:
-                $ref: '#/components/schemas/Pet'
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Pet'
-        '400':
-          description: Invalid ID supplied
-        '404':
-          description: Pet not found
-        '405':
-          description: Validation exception
-      security:
-        - petstore_auth:
-            - 'write:pets'
-            - 'read:pets'
-      requestBody:
-        $ref: '#/components/requestBodies/Pet'
-  /pet/findByStatus:
+  /records:
     get:
       tags:
-        - pet
-      summary: Finds Pets by status
-      description: Multiple status values can be provided with comma separated strings
-      operationId: findPetsByStatus
-      parameters:
-        - name: status
-          in: query
-          description: Status values that need to be considered for filter
-          required: true
-          style: form
-          explode: false
-          deprecated: true
-          schema:
-            type: array
-            items:
-              type: string
-              enum:
-                - available
-                - pending
-                - sold
-              default: available
+        - website-record
+      summary: Get all website records
+      description: Gets a list of all the current website records
+      operationId: getRecordsList
```

Comment:
This endpoint (`GET /api/records/`, i.e. retrieving the list of all records) should have some search and sorting - see the snippet below. 

![obrazek](https://github.com/barjin/nswi153-crawler/assets/61918049/640177a3-4103-4f1f-b1be-176ed6ccb1cc)

I'd add two query parameters for this: `q` and `sort` (or similar names).

Examples (search):
- searching for all the records with label 'test':
  -  `GET /api/records?q=label:test`
- searching for all the records with tags `a` and `b`:
  - `GET /api/records?q=tags:a+b`
  
The problem is that we could filter only by one field at a time (only by `label` or `tags` or `URL`). Alternatively, we could have multiple query parameters (`?label=`, `?tags=` and `?url=`) that would all serve as filters for their respective fields (now that I'm thinking about this, I like this more 😅 ).

Examples (sort):
- sorting the records by URL:
  - `GET /api/records?sort=url:desc` etc.

Again, we could split this into `sortField` and `sortDirection` - I'm undecided on this.

Reply:
I went with separate parameters for each query filter and a single parameter for sort